src/Compiler/Scheme/Common.idr

```diff
@@ -86,21 +86,60 @@ shl _                       x y = op "blodwen-shl" [x, y]
 
 castInt : Constant -> Constant -> String -> String
```

Comment:
I tried to factor out the common patterns as suggested. Note that we still need to pass `IntKind`s to the backends, since the JS backend will need this information to decide, which representation to use for an integer (once we distinguish between `BigInt` and `Number`). There are certainly more common patterns that could be packed into `ConstantPrimitives` (arithmetic operations come to mind), but right now I'd rather leave this as it is. I'll happily try to factor out some more stuff in future PRs.

Reply:
Ah I see. I thought that all of the business of
`if m >= n then x else someformoftruncation x`
could be handled once and for all but that does
assume all the numbers are represented the same way.

Let's go with the current version then.